core/modules/url/helpers/index.ts

```diff
@@ -57,7 +62,8 @@ export function findRouteByPath (path: string): RouteConfig {
 export function normalizeUrlPath (url: string): string {
   if (url && url.length > 0) {
     if (url.length > 0 && !url.startsWith('/')) url = `/${url}`
-    if (url.endsWith('/')) url = url.slice(0, -1)
+    if (url.endsWith('/') && !config.seo.urlTrailingSlash) url = url.slice(0, -1)
+    if (!url.endsWith('/') && config.seo.urlTrailingSlash) url += '/'
     const queryPos = url.indexOf('?')
     if (queryPos > 0) url = url.slice(0, queryPos)
   }
```

Comment:
Last changes in different file:
1. https://github.com/DivanteLtd/vue-storefront/blob/master/core/modules/url/router/beforeEach.ts#L26
change `const path = normalizeUrlPath(to.path)` to `const path = normalizeUrlPath(to.path, false)` => in this place we removed trailing slash, so we need to use `normalizeUrlPath` with `false` which will result in keeping trailing slash
2. https://github.com/DivanteLtd/vue-storefront/blob/master/core/modules/url/router/beforeEach.ts#L33
change `if (routeData) {` to `if (routeData && !routeData.name.endsWith('page-not-found')) {`